platform/linux-generic/odp_schedule.c

```diff
@@ -1141,9 +1143,17 @@ static void schedule_order_unlock(unsigned lock_index)
 
 	ODP_ASSERT(sched_local.ordered.ctx == odp_atomic_load_u64(ord_lock));
 
+	sched_local.ordered.lock_index = sched->queue[queue_index].
+					order_lock_count + 1;
```

Comment:
Nice way to avoid the extraneous checking in `schedule_order_unlock_lock()`, but actually this isn't needed at all due to the prior `ODP_ASSERT()`. That assert requires that the `lock_index` lock be held at entry to `schedule_order_unlock` so if `odp_schedule_order_unlock_unlock()` is called without an ordered lock being held, the previous `ODP_ASSERT()` will detect that and abort. 

Interestingly, this also means that (in odp-linux, at least) you can call `odp_schedule_order_unlock_lock()` while holding nested ordered locks. The semantics is always to release the most recently acquired ordered lock and replace that with a new ordered lock regardless of how many other ordered locks the caller holds in this ordered context.

So the net is you can delete this statement without any loss of correctness.

Reply:
I added this change for scenario where odp_schedule_order_unlock_lock() is called after a lock has been released using odp_schedule_order_unlock() but since we are defining in the documentation this is an undefined behaviour I believe your suggestion makes sense.
I will update in the next version.